### PR TITLE
No crash when error code in partition metata data

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 * 1.5.7
   - Stop supervised producer if failed to start. Otherwise the caller may have to call the wolff:stop_and_delete_supervised_producers/3
-    after matching an error return. If they don't, then it may appear as a resource leak.
+    after matching an error return. If they don't, then it may appear as a resource leak. [#26](https://github.com/kafka4beam/wolff/pull/26)
+  - Ensure `{{Topic, Partitio}, Connection}` record exists even if there are errors returned at partition level metadata.
```
Comment (ieQu1) [inline changelog.md:4]:
```suggestion
  - Ensure `{{Topic, Partition}, Connection}` record exists even if there are errors returned at partition level metadata.
```

File: changelog.md [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+    Fixed in PR [#29](https://github.com/kafka4beam/wolff/pull/29).
+    There were two issues before this fix:
+    * `wolff_client` may crash when trying to find partition leader connection for a producer worker.
+      When there is error code in partition metadata, the connection record is not added,
+      causing a `badmatch` error in this expression `{_, MaybePid} = lists:keyfind(Partition, 1, Partitions)`.
+    * `wolff_producers` may crash when fewer partitions found in partition counter refresh.
+      Although Kakfa does not support topic down-scale, the assertion has been removed.
```
Comment (ieQu1) [inline changelog.md:11]:
```suggestion
      Although Kafka does not support topic down-scale, the assertion has been removed.
```

File: src/wolff_client.erl [hunk 3]
```diff
@@ -314,8 +340,8 @@ is_connected(MaybePid, {Host, Port}) ->
 
 is_alive(Pid) -> is_pid(Pid) andalso erlang:is_process_alive(Pid).
 
-add_conn({ok, Pid}, ConnId, Conns) -> Conns#{ConnId => Pid};
-add_conn({error, Reason}, ConnId, Conns) -> Conns#{ConnId => Reason}.
+add_conn({ok, Pid}, ConnId, Conns) when is_pid(Pid) -> Conns#{ConnId => Pid};
+add_conn({error, Reason}, ConnId, Conns) when not is_pid(Reason) -> Conns#{ConnId => Reason}.
```
Comment (ieQu1) [inline src/wolff_client.erl:344]: Is guard expression really needed here? Seems a bit strange for the reason to be a Pid, and this case is not handled either.

Comment (zmstone) [reply src/wolff_client.erl:344]: it's intended to crash when Reason is actually a pid

**Pull request merged**